### ISSUE TITLE
Support S3 private bucket repository from helm deployment

### ISSUE THREAD

**Issue by qinxx108** [aws/aws-cdk#9273]
### Use Case
Right now helm deployment only support public github repository. We need to read the repository from s3 private repo

### Proposed Solution

> 
I would like to help to develop the feature but would like to hear your opinion on the approach

1. Create a s3 presign url and use the presign url to deploy the chart
`new HelmChart(this, 'cortexHelm', {
      cluster: props.cluster,
      chart: props.presignUrl,
      namespace: props.environment.cluster.namespace,
      values: overrides,
      release: 'cortex'
    });`

2. install S3 plugin at lambda layer: https://github.com/aws-samples/aws-lambda-layer-kubectl
Use the S3 plugin to read from S3 private repo while deploy helm chart: 
Plugin link: https://github.com/hypnoglow/helm-s3

### Other

* [x] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

**Comment by qinxx108** [comment]
Hello, i want to submit a PR for this implementation but i would like to hear your opinion about which one that your team prefer. Please take a look at the description 

Thanks!
yijie

**Comment by iliapolo** [comment]
@qinxx108 Could you please give some more detail on the first approach of using presign URL's? In general, an approach that doesn't require a custom lambda layer is definitely preferable - but i'd like to understand exactly what it entails and what will be the flow from a user perspective

**Comment by eladb** [comment]
@qinxx108 and me chatted today and a potential approach. @qinxx108 will update this issue with details.

**Comment by polothy** [comment]
Not sure if this makes anything easier, but ECR now support OCI: https://aws.amazon.com/blogs/containers/oci-artifact-support-in-amazon-ecr/

Helm can then deploy a chart that is in ECR.

**Comment by jmjava** [comment]
@polothy is there a CDK construct that supports ECR - have you tried this?  I have tried using the helmChart construct cannot get it to work.  Are you saying use helm directly like in a lambda from CDK?   We have others in our group that are not using CDK but are using s3 static website that has IP range locked down to make it "private"  - I think this may be our best option to keep CDK code clean at the moment

**Comment by polothy** [comment]
No, I haven't tried it.  And yeah, think you would need to run helm in a lambda, which I think they do for the regular helm installs.

**Comment by plumdog** [comment]
I have opened a PR https://github.com/aws/aws-cdk/pull/13496 in which I accidentally reinvented what @nicolai86 suggested, but with a less nice API and more `subprocess`. I am hopeful this will be a useful step towards handling resolving this issue, but it doesn't solve the original issue as described. I think it might solve the intent of the issue though.
